lib/balance-tx/lib/Cardano/Tx/Balance/Internal/CoinSelection.hs

```diff
@@ -263,18 +263,12 @@ toInternalSelectionConstraints SelectionConstraints {..} =
 -- | Specifies all parameters that are specific to a given selection.
 --
 data SelectionParams = SelectionParams
-    { assetsToBurn
-        :: !TokenMap
-        -- ^ Specifies a set of assets to burn.
-    , assetsToMint
-        :: !TokenMap
-        -- ^ Specifies a set of assets to mint.
```

Comment:
@jonathanknowles  Please help me to understand the removal of these fields.
1. Why coin selection parameters needed to know about assets to mint and assets to burn before?
2. Why is this knowledge isn't represented in the rhs of this change?
Thank you!

Reply:
> Why coin selection parameters needed to know about assets to mint and assets to burn before?

It is similar to https://github.com/input-output-hk/cardano-wallet/pull/3926

Before balanceTx, coin-selection needed to compute the balance of the tx/selection on its own. Whether or not the tx contained assets to mint or burn was then highly relevant. To make `balanceTx` possible we "hijacked" these fields to control coin-selection's notion of balance (cf e.g. https://github.com/input-output-hk/cardano-wallet/pull/3093#issuecomment-1020406969). Now `balanceTx` is the only caller, and the `assetsToMint` fields are _always_ used as "extra balance" and never as "assetsToMint". So it can be renamed/simplified.

Reply:
> Please help me to understand the removal of these fields.

Sure, happy to try and explain!

> 1. Why coin selection parameters needed to know about assets to mint and assets to burn before?

1. In the past, before the creation of our grand unified `balanceTx` function, our main tool for balancing transactions was the coin selection algorithm. We typically called into coin selection via the `selectAssets` function of `Cardano.Wallet`.
2. At some point, it became a requirement for the wallet to support the creation of transactions that mint or burn tokens.
3. Since the coin selection algorithm already had the knowledge of how to balance transactions, and since we judged that it would be relatively simple to upgrade the algorithm to support minting and burning, we took the decision to make the coin selection algorithm responsible for this.
4. The `extraCoinIn` and `extraCoinOut` fields were added to support reward withdrawals and certificate deposits.

> 2. Why is this knowledge isn't represented in the rhs of this change?

I'm not sure what you mean by "the RHS of this change". Could you elaborate? (Happy to explain further.)